caddytls/ctmonitor.go

```diff
@@ -0,0 +1,298 @@
+package caddytls
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mholt/caddy"
+)
+
+func init() {
+	caddy.RegisterEventHook("ctmonitor", startMonitoring)
+}
+
+const (
+	certSpotterAPIBase string = "https://api.certspotter.com/v1/issuances"
+)
+
+var (
+	filePath     string = filepath.Join(caddy.AssetsPath(), "ct_id")
+	ctConfigFile string = filepath.Join(caddy.AssetsPath(), "ct_config")
+)
+
+// CtConfig struct will allow me to get the config data from a file
+type CtConfig struct {
+	IncludeSubdomains bool `json:"subdomains"`
+	IncludeWildCards  bool `json:"wildCards"`
+}
+
+// CertSpotterResponse is the json response formatting that is used in the program.
+type CertSpotterResponse struct {
+	ID           string   `json:"id"`
+	TBSSHA256    string   `json:"tbs_sha256"`
+	DNSNames     []string `json:"dns_names"`
+	PubKeySha256 string   `json:"pubkey_sha256"`
+	Issuer       struct {
+		Name         string `json:"name"`
+		PubKeySha256 string `json:"pubkey_sha256"`
+	} `json:"issuer"`
+	NotBefore string `json:"not_before"`
+	NotAfter  string `json:"not_after"`
+	Cert      struct {
+		ID     string `json:"id"`
+		Type   string `json:"type"`
+		SHA256 string `json:"sha256"`
+		Data   string `json:"data"`
+	} `json:"cert"`
+}
+
+// QueryConfig is the configuration information for my queries.
+type QueryConfig struct {
+	Subdomains bool
+	WildCards  bool
```

Comment:
Instead of repeating these fields, consider embedding the `CTConfig` struct (look up embedded structs).